test/hpke_test.c

```diff
@@ -0,0 +1,1410 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/evp.h>
+#include <openssl/core_names.h>
+#include <openssl/rand.h>
+#include <openssl/hpke.h>
+#include "testutil.h"
+
+#ifndef OSSL_HPKE_MAXSIZE
+# define OSSL_HPKE_MAXSIZE 512
+#endif
+
+static OSSL_LIB_CTX *testctx = NULL;
+static char *testpropq = NULL;
+
+/**
+ * @brief compare an EVP_PKEY to buffer representations of that
```

Comment:
Test that an EVP_PKEY encoded public key matches the supplied buffer.

Reply:
sure